### PR TITLE
Add parameter permissions to control panel

### PR DESCRIPTION
**Pull request by s-block**
Gives the control panel ability to create parameters

## What

Changed the IAM policy to add statement for creating Params

### CODE REVIEW

**Comment by xoen** [comment]
Don't know if we want to keep prefixing resource names by environment to logically separate them, e.g `dev` CP has only permission to touch `"arn:aws:ssm::${var.account_id}:parameter/dev_*"` params and `alpha` CP has only permission to touch `"arn:aws:ssm::${var.account_id}:parameter/alpha_*"` (it would avoid potential problems).

Otherwise looks good from the little I know about AWS parameter store 👍

File: infra/terraform/modules/control_panel_api/role.tf [hunk 1]
```diff
@@ -148,6 +148,21 @@ resource "aws_iam_policy" "control_panel_api" {
...
+      "Action": [
+        "ssm:PutParameter",
+        "ssm:DeleteParameter",
+        "ssm:GetParameterHistory",
+        "ssm:GetParametersByPath",
+        "ssm:DeleteParameter",
+        "ssm:DeleteParameters"
+      ],
+      "Resource": [
+        "arn:aws:ssm::${var.account_id}:parameter/*"
```
Comment (r4vi) [inline infra/terraform/modules/control_panel_api/role.tf]: good point by @xoen about namespacing the params

File: infra/terraform/modules/control_panel_api/role.tf [hunk 2]
```diff
@@ -148,6 +148,31 @@ resource "aws_iam_policy" "control_panel_api" {
...
       ]
     },
+    {
+      "Sid": "CanCreateAndDeleteSSMParameters",
+      "Effect": "Allow",
+      "Action": [
+        "ssm:PutParameter",
+        "ssm:DeleteParameter",
+        "ssm:GetParameterHistory",
+        "ssm:GetParametersByPath",
```
Comment (andyhd) [inline infra/terraform/modules/control_panel_api/role.tf]: Do we need `GetParameterHistory`? And do we not need `GetParameter`?

**Comment by r4vi** [comment]
I would've thought cpanel should only set them and not be able to retrieve
the value

File: infra/terraform/modules/control_panel_api/role.tf (hunk 2 above)
Comment (r4vi) [reply infra/terraform/modules/control_panel_api/role.tf]: I would've thought cpanel should only set them and not be able to retrieve
the value

Comment (s-block) [reply infra/terraform/modules/control_panel_api/role.tf]: Yeah, there was something about listing in boto which required history. But no, we don't want to get the parameter in control panel

**Pull request merged**